Assets/MRTK/SDK/Features/Utilities/Migration/Tools/MigrationTool.cs

```diff
@@ -44,33 +45,112 @@ public MigrationTool()
+
+                if (!SetMigrationHandlerInstance(type))
+                {
+                    return false;
+                }
+            }
+
             if (!selectedObject)
             {
-                Debug.LogError("Selection is Empty. Could not add for migration.");
+                Debug.LogError("Selection is empty. Please select object for migration.");
                 return false;
             }
 
             if (selectedObject is GameObject || selectedObject is SceneAsset)
             {
-                if (!migrationObjects.Contains(selectedObject))
+                if (CheckIfCanMigrate(type, selectedObject) && !migrationObjects.Contains(selectedObject))
                 {
                     migrationObjects.Add(selectedObject);
-                    Debug.Log($"{selectedObject.name} object added for migration");
+                    return true;
+                }
+                else
+                {
+                    Debug.Log($"{selectedObject.name} does not support {type.Name} migration. Could not add object for migration");
+                    return false;
                 }
-                return true;
             }
             Debug.LogError("Object must be a GameObject, Prefab or SceneAsset. Could not add object for migration");
             return false;
         }
 
+        private bool CheckIfCanMigrate(Type type, Object selectedObject)
+        {
+            string objectPath = AssetDatabase.GetAssetPath(selectedObject);
+
+            // Object migrationObject is a scene game object
+            if (String.IsNullOrEmpty(objectPath) && selectedObject is GameObject)
+            {
+                var objectHierarchy = ((GameObject)selectedObject).GetComponentsInChildren<Transform>();
+                for (int i = 0; i < objectHierarchy.Length; i++)
+                {
+                    try
+                    {
+                        if (migrationHandlerInstance.CanMigrate(objectHierarchy[i].gameObject))
+                        {
+                            return true;
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        Debug.LogError($"{e.Message}: Could not check if GameObject {objectHierarchy[i].name} can be migrated");
+                    }
+                }
+            }
+            else
+            {
+                // Selected object is prefab asset
+                if (selectedObject is GameObject)
+                {
+                    PrefabAssetType prefabType = PrefabUtility.GetPrefabAssetType(selectedObject);
+                    if (prefabType == PrefabAssetType.Regular || prefabType == PrefabAssetType.Variant)
+                    {
+                        var parent = UnityEditor.PrefabUtility.LoadPrefabContents(objectPath);
+
+                        if (CheckIfCanMigrate(type, parent))
```

Comment:
Nit: Could also consider having it like:


```c#
var canMigrate = CheckIfCanMigrate(type, parent);
PrefabUtility.UnloadPrefabContents(parent);
return canMigrate;
```

So that you don't have to maintain cleanup in two different branches - in this case it wont' be an issue going forward but in general having duplicate cleanup paths can be super dangerous (you might update one and forget the other)